plugins/suspendmanager.cpp

```diff
@@ -291,6 +298,70 @@ class SuspendManager {
         return false;
     }
 
+    static bool suitableToMoveItemTo(df::coord pos) {
+        if (!walkable(pos))
+            return false;
+
+        // Dwarves will move a blocking item to a walkable building, but only if its fully constructed
+        auto building = Buildings::findAtTile(pos);
+        return !building || building->flags.bits.exists;
+    }
+
+    // True for buildings planned on a tile with an item that cannot be moved aside
+    static bool isOnUnmovableItem(df::job *job) {
+        CHECK_NULL_POINTER(job);
+        auto building = Job::getHolder(job);
+        if (!building)
+            return false;
+
+        // Check for items on the building, look into all the map blocks overlapping
+        // the building
+        bool has_item = false;
+        for (int32_t block_x = building->x1 >> 4; block_x <= building->x2 >> 4; ++block_x) {
+            for (int32_t block_y = building->y1 >> 4; block_y <= building->y2 >> 4; ++block_y) {
+                auto block = Maps::getBlock(block_x,block_y,building->z);
```

Comment:
this would be easier to understand, I think, if you didn't bitshift the coordinates and used `Maps::getTileBlock` instead of `Maps::getBlock`

Reply:
I'm annoyed by that too, but the issue is that the list of items is stored on the map block, and I did not find any direct access from a tile to an item. Due to that, if the iteration is on all the building tile, this would multiply the checks to do (from the tile find the map block, from the map block enumerate the items and find the ones on the tile). A worst case scenario would be a 4x4 building on a single map block, enumerating the items in this block 16 times instead of one :(

Reply:
ah ok, understood